pkg/virt-controller/watch/vmi.go

```diff
@@ -1196,7 +1196,26 @@ func (c *VMIController) addVirtualMachineInstance(obj interface{}) {
 
 func (c *VMIController) deleteVirtualMachineInstance(obj interface{}) {
 	c.lowerVMIExpectation(obj)
```

Comment:
oh, now I see. We need t move `c.lowerVMIExpectation()` to the end too and pass in the VMI.